app/api/v2/identity/users.rb

```diff
@@ -5,6 +5,20 @@
 module API::V2
   module Identity
     class Users < Grape::API
+      helpers do
+        def validate_refid?
+          return false if params[:refid].nil?
+
+          error!('Invalid referral uid format', 422) unless /\AID\w{10}$/.match?(params[:refid])
+          error!("Referral doesn't exist", 422) if User.find_by_uid(params[:refid]).nil?
+          true
+        end
+
+        def referral_id
+          User.find_by_uid(params[:refid]).id
```

Comment:
what happens if it doesn't find UID ?

Reply:
try to make functions bigger than 1 lines

Reply:
```
if User.find_by_uid(params[:refid]).nil?
error!("Referral doesn't exist", 422)
```

Squashed 2 funcs into one